### DISCUSSION TITLE
Version 1.0.7

### DISCUSSION THREAD

**Discussion by ABaumher** [ABaumher/galaxy-integration-steam#31] [Announcements]
Dropping the beta tag because it's gone through excessive testing and the issues we're running into won't likely be fixed without the major 2.0 rework. 

This version fixes a long-standing issue with accounts that had Steam Guard *disabled*. They were never noticed, likely because Steam very much prefers you use 2FA, but while tracking down some other code a discrepency was noticed and it was then fixed. 

A small Quality of Life change has been added: For those of you who use 2FA, any leading or trailing whitespace is now removed. Usually when you copy and paste the code from email, it'd contain a trailing whitespace character. 

As per usual, there are 2 versions: one with an installer, and a standalone release. MacOS will be released when we can. 

If you are on version 1.0.5 or newer, this version should be installed automatically. Galaxy Client does some voodoo magic and i'm not sure *how* but it just works. 

<hr /><em>This discussion was created from the release <a href='https://github.com/ABaumher/galaxy-integration-steam/releases/tag/v1.0.7'>Version 1.0.7</a>.</em>

**Comment by ABaumher** [comment]
@urwrstkn8mare or @bgebhardt i could use a 1.0.7 mac version when you get a chance.
@urwrstkn8mare you can do this by running `inv pack` from your virtual environment. you will of course need to active the environment first. 

If you post it here or email it to me if you think that's safer, i can tar it from WSL with the macos install script and hopefully keep the permissions. 

I'm hoping this is the final release before 2.0

**Reply by urwrstkn8mare** [reply]
[steam_v1.0.7.zip](https://github.com/ABaumher/galaxy-integration-steam/files/11796658/steam_v1.0.7.zip)

**Reply by ABaumher** [reply]
Ty

**Reply by ABaumher** [reply]
@urwrstkn8mare can you let me know if the command in the `mac_with_installer.tar.gz` has execute permission? It seems to lose permissions if not compressed before being uploaded/downloaded, and uploading it is really finicky on windows (which does not have permissions). i used the `tar` command on WSL Ubuntu and `ls -la` showed full execute permission before i zipped it in.

**Reply by urwrstkn8mare** [reply]
yep it got the perms 👍

**Reply by ikkinomi** [reply]
thank a lot, working well on windows 11 !!

**Comment by yoharnu** [comment]
windows.zip manifest.json is reporting v1.0.6. Unsure if the manifest is wrong or if the wrong files are packaged.

**Reply by urwrstkn8mare** [reply]
`src/version.py` still has `1.0.6` maybe thats it

**Reply by ABaumher** [reply]
I'll check it and fix it if needed

**Reply by ABaumher** [reply]
wrong files. pushing a new one now

**Comment by teh1archon** [comment]
Hi, I've updated to 1.0.7 from the old 1.0.2 and I use Steam Guard (not 2FA via mail or something). The plugin connects and stays connected but it fails to import my Steam Library. It goes to 98% gets stuck there for several minutes, then it gets stuck on "importing steam..." and then... nothing. When I click on the Steam tab in GOG it's empty with 0 games. 
Help :(

**Reply by ABaumher** [reply]
SteamGuard is valve's fancy term for 2FA. So unless you disable steam guard you have 2fa, just an fyi. The terms can be confusing, frankly I'm not sure why valve chose a special term here. 

We're going to need more information that just "it's not working" to help you out. The best way to do that is the log file. The main page of the repo has the location where the logs are kept. Post it here and we can read it and figure out what was going on when the plugin was running. 

In the meantime, some simple questions - did you need to log in before it connected? How big is your library? Did you try closing and reopening gog galaxy (there's a known issue that it might not sync properly the first run, but restating seems to fix it. It's been annoyingly difficult to track down)?

**Reply by teh1archon** [reply]
Thanks for the quick reply. I did have to log in as I disconnected before I deleted version 1.0.2 of the plugin going forward with 1.0.7.  I also deleted the local cache in one of my attempts to fix it. I closed and re-opened GOG several times.
Before my total owned games was about 9000 now it's about 5000, the Humble Bundle plugin also stopped working (among others) so the reading of unredeemed keys is somewhat off and outdated. My page on the Steam client s
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/11923593/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)
ays I have 3,290 games.
I don't know if something "confusing" GOG with some games in my library are on several platforms (PSN, GOG, Origin, Uplay, Humble, Itch, Epic). Never had a problem with this before.
I see in the logs folder that it keeps writing and trying to log. Attached are the 2 latest logs.
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8 (2).log](https://github.com/ABaumher/galaxy-integration-steam/files/11923596/plugin-steam-ca27391f-2675-49b1-92c
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/11923607/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)
0-896d43afa4f8.2.log)


I'm also usually behind a VPN (NordVPN), and it gives me trouble from time to time (recently specifically with Epic and Humble) but never with GOG and never with the integrations unless it or something else just disconnected me from them, but had no problem to re-establish the connection.

**Reply by ABaumher** [reply]
looking through your logs, we spend most of that time idle - waiting, essentially. It seems odd. If i had to hazard a guess, we're losing a response, but i can't confirm it. Also, we're using TCP so we shouldn't lose messages but it's possible if a connection drops (and we restart a new one) while we're getting a message we'll wait for it forever. 

Just to double check - can you delete your logs, then try again? I want a set of fresh logs that i can track the workflow. If the plugin logs a lot of info, it may create .0 .1, etc files for the steam plugin log, so include those as well. Finally, there should be a `GalaxyClient.log` as well, post that here just in case it's useful.

If that's the case, it's something i've been looking to address in my 2.0 rework, which is getting closer to being a thing. We can properly handle things that are waiting on messages when we lose a connection - essentially, wait until the new connection is started then send the message over.

**Reply by teh1archon** [reply]
OK, I cleared the logs and opened GOG again fresh. Still was stuck for like an hour or so on "Importing Steam...", when it gave up I closed GOG. These are the logs:
[GalaxyClient.log](https://github.com/ABaumher/galaxy-integration-steam/files/11960513/GalaxyClient.log)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/11960515/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

Good luck :)

**Reply by ElemWiz** [reply]
I've been having the exact same problem as teh1archon. but I'm not using a VPN. Had the same problem with that Humble Bundle integration too, so deactivated it, otherwise it just sat there trying to import HB seemingly forever. Unfortunately, it sits there trying to import Steam with nothing to show for it either (and I have a measly 831 games compared to teh1archon's mammoth amount). Hope you guys manage to fix the problem. Should I close GOG, delete the existing Steam logs, load GOG back up, and then send the new log too?

**Reply by ABaumher** [reply]
Unfortunately, i now have the same issue. For me, closing and reopening fixed it but I'm not sure why. 
Unfortunately bugs i can't reproduce are annoying to track down. I actually added a bunch of log statements to try to track it down... and then it worked immediately. 

What i can recommend is trying the 1.2 Beta. If the issue is due to quirky code from protobuf, it's been addressed. Or any other number of issues that have been fixed. 

Without giving you all too much hope (i'm still very much burned out of this thing) the reason i went back into gog was to try to demystify the responses steam gives us so we can get 2.0 finished. It's giving me more questions than answers (every response is packed in a zip, even if there's only one of them), but getting all of that understood will help make everything smoother

**Reply by ABaumher** [reply]
Follow up - did either of you accept a friend request or add games before this issue popped up? Retracing my steps, i accepted a friend request and got 2 games before i tried it the first time

**Reply by ElemWiz** [reply]
I didn't accept any friend requests, but it looks like updating to the 1.2 Beta fixed the problem. It just takes a lil while - which isn't really a big deal for me because it works. It even has my newer purchases. :-)

**Reply by ElemWiz** [reply]
Even still has all my created tags for each of the games.

**Reply by teh1archon** [reply]
I will try later to switch to 1.2 beta, I wasn't gifted any games (not that I remember), and I had incoming friend requests that I've ignored (potential leechers trying to get unclaimed keys from HB) but I have a feeling that was not it. If adding specific games is an issue, I'd bet it's one of two things:
1. The Orange Box that was like 96% off and all the games were added to my account (some with delay) except for Team Fortress 2 that it seems is a known problem not showing in the library (I took it when it was free for a limited time before going F2P like 10 years ago). Since then to play I must press on Play from the store page if no desktop shortcut is there.
2. The new option that you can add games to your library without starting to install them first. I'm pretty sure my issues started when I started to add free games to my library but I don't remember for sure if that was the turning point it stopped updating altogether. I know that in Itch.io case it shows me only games that I bought with money, claiming games that 100% won't show. Maybe there's a connection? No idea, not even sure it's a good lead.

Thank you very much ABaumher for still being on the case, I really appreciate it.

**Reply by teh1archon** [reply]
I can confirm that 1.2 beta works but it failed to add 20+ games that I've added to my Steam library in the last few days. Moreover, it seems also to fail to update itself with Epic games for some reason. 
Well, it's better than nothing but too bad that now when I need to check if I own a game, just using GOG Galaxy I still need to check Epic, Humble, and Steam as well manually. But pretty sure this is an issue with GOG at this point as far as it concerns Epic... 

Thanks ABaumher :)

**Reply by teh1archon** [reply]
I've updated GOG right now and the new games from Steam were detected :) Huzzah!

**Comment by EdoardoMolinatti** [comment]
Hi guys, this works like a charm on Windows, thank you!
It was utterly annoying to have an error every time I started GOG Galaxy, and in the end, it didn't connect to Steam.

It's all terrific, just one obvious thing to do for the Windows version (and probably Mac too):
- Please check if GOG is running, or at least print advice for the user to **exit from GOG** before installing the plugin with the script `auto-copy-windows.bat` (or the Mac counterpart).

I know this is quite an obvious thing to do (to exit from GOG), but I was distracted and it didn't work the first time (many files are locked while GOG is running, and there are many errors on the terminal in that case). Of course, it works like a charm once GOG is closed.
It would be a user-friendly feature (or just a warning), for newbies or distracted users.

Thank you again for the awesome work!

**Comment by T1Cybernetic** [comment]
Worked perfectly :) Thank you guys very much for the very much needed fix!
